common/shared.c

```diff
@@ -86,6 +86,7 @@ static int getaddr(char *node, char *service,
 {
 	int ret;
 	struct fi_info *fi;
+	struct fi_info hcopy = *hints;
```

Comment:
IMO, if hints are changed, I would consider it a bug.  We should update the API to define it as const.